src/emulator/storage/persistence.ts

```diff
@@ -74,10 +82,19 @@ export class Persistence {
   }
 
   renameFile(oldName: string, newName: string): void {
+    this._diskPathMap.set(newName, uuid.v4());
     renameSync(this.getDiskPath(oldName), this.getDiskPath(newName));
   }
 
   getDiskPath(fileName: string): string {
+    const shortenedDiskPath = this._diskPathMap.get(fileName);
+    if (shortenedDiskPath !== undefined) {
+      return path.join(this._dirPath, encodeURIComponent(shortenedDiskPath));
+    }
     return path.join(this._dirPath, encodeURIComponent(fileName));
   }
+
+  addDiskPathMapping(fileName: string, diskPath: string): void {
```

Comment:
I see this disk path map as an implementation detail of this class. Is there a way we can avoid exposing this to consumers of this class?

Reply:
Will remove this, found a workaround so I don't need to expose it